### PR TITLE
[gitlab-owners] ignore 0 commits MRs and run in parallel for all repos

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-2768

### CODE REVIEW

File: reconcile/gitlab_owners.py [hunk 1]
```diff
@@ -32,8 +34,13 @@ def __init__(self, gitlab_client, merge_request, owners, dry_run):
         self.owners = owners
         self.dry_run = dry_run
 
-        top_commit = next(self.mr.commits())
-        self.top_commit_created_at = dateparser.parse(top_commit.created_at)
+        commits = self.mr.commits()
+        if len(commits) != 0:
```
Comment (apahim) [inline reconcile/gitlab_owners.py:38]: `if commits`

File: reconcile/gitlab_owners.py [hunk 2]
```diff
@@ -32,8 +34,13 @@ def __init__(self, gitlab_client, merge_request, owners, dry_run):
...
+            top_commit = next(self.mr.commits())
```
Comment (apahim) [inline reconcile/gitlab_owners.py:39]: now you have `commits`... no need to call `mr.commits()` again, I guess

**Review by apahim — APPROVED** [review]
lgtm, thanks for this

File: reconcile/gitlab_owners.py (hunk 1 above)
Comment (maorfr) [reply reconcile/gitlab_owners.py]: :+1:

File: reconcile/gitlab_owners.py (hunk 2 above)
Comment (maorfr) [reply reconcile/gitlab_owners.py]: too early in the morning

**Pull request merged**